src-v2/NTS.Domain.Core/Aggregates/Participations/Participation.cs

```diff
@@ -70,7 +74,18 @@ public void Update(IPhaseState state)
         phase.Update(state);
         EvaluatePhase(phase);
     }
-
+    public void IsReinspectionRequested(bool requested)
+    {
+        GuardHelper.ThrowIfDefault(Phases);
+        GuardHelper.ThrowIfDefault(Phases.Current);
```

Comment:
Current can be null, but this is not a good place for a GuardException. Any idea why?